apps/src/netsim/NetSimSlider.js

```diff
@@ -0,0 +1,338 @@
+  if (pos > this.valueToSliderPosition(this.maxValue_)) {
+    return this.isUpperBoundInfinite_ ? Infinity : this.maxValue_;
+  } else if (pos < this.valueToSliderPosition(this.minValue_)) {
+    return this.isLowerBoundInfinite_ ? -Infinity : this.minValue_;
+  }
+  return pos;
+};
+
+/**
+ * Converts an external-facing numeric value into a localized string
+ * representation of that value.
+ * @param {number} val - numeric value of the control
+ * @returns {string} - localized string representation of value
+ */
+NetSimSlider.prototype.valueToLabel = function (val) {
+  if (val === Infinity || val === -Infinity) {
+    return i18n.unlimited();
+  }
+  return val;
+};
+
+/**
+ * Alternate label converter, used for slider end labels.
+ * @param {number} val - numeric value of the control
+ * @returns {string} - localized string representation of value
+ */
+NetSimSlider.prototype.valueToShortLabel = function (val) {
+  return this.valueToLabel(val);
+};
+
+/**
+ * Default minimum of zero is useless to a logarithmic scale
+ * @type {number}
+ * @const
+ */
+var LOGARITHMIC_DEFAULT_MIN_VALUE = 1;
+
+/**
+ * By default, a logarithmic scale slider increases by a factor of 2
+ * every step.
+ * @type {number}
+ */
+var LOGARITHMIC_DEFAULT_BASE = 2;
+
+/**
+ * @param {jQuery} rootDiv
+ * @param {Object} options - takes NetSimSlider options, and:
+ * @param {number} options.logBase - factor by which the value increases
+ *        with every slider step.  Default base 2.
+ * @constructor
+ * @augments NetSimSlider
+ */
+NetSimSlider.LogarithmicSlider = function (rootDiv, options) {
+  options.min = utils.valueOr(options.min, LOGARITHMIC_DEFAULT_MIN_VALUE);
+  NetSimSlider.call(this, rootDiv, options);
+
+  /**
+   * Factor by which the value increases with every slider step.
+   * @type {number}
+   * @private
+   */
+  this.logBase_ = utils.valueOr(options.logBase, LOGARITHMIC_DEFAULT_BASE);
+
+  /**
+   * Precalculate natural log of our base value, because we'll use it a lot.
+   * @type {number}
+   * @private
+   */
+  this.lnLogBase_ = Math.log(this.logBase_);
+
+  this.calculateSliderBounds_();
+};
+NetSimSlider.LogarithmicSlider.inherits(NetSimSlider);
+
+NetSimSlider.LogarithmicSlider.prototype.calculateSliderBounds_ = function () {
+  // Pick boundary slider values
+  this.maxSliderPosition = this.logFloor_(this.maxValue_);
+  // Add a step if we don't already land exactly on a step, to
+  // compensate for the floor() operation
+  if (Math.pow(this.logBase_, this.maxSliderPosition) !== this.maxValue_) {
+    this.maxSliderPosition += this.step_;
+  }
+  this.minSliderPosition = this.logFloor_(this.minValue_);
+
+  // Pick infinity slider values
+  this.infinitySliderPosition = this.maxSliderPosition + this.step_;
+  this.negInfinitySliderPosition = this.minSliderPosition - this.step_;
+};
+
+NetSimSlider.LogarithmicSlider.prototype.logFloor_ = function (val) {
+  var ceilThreshold = 0.0000001;
```

Comment:
Hmmm....If these numbers were more important to us, I'd say it might be worth using some sort of math library. Given that this is just for a slider, your approach is probably fine. I might add a little comment about the magic number though.